### ISSUE TITLE
Shell()() should throw an exception if the launch immediately fails

### ISSUE THREAD

**Issue by pjknowles** [molpro/sjef#221]

**Comment by nick-wilson** [comment]
Added code to throw exceptions in asynchronous and synchronous cases in:
https://github.com/molpro/sjef/commit/3e48572f55131d2277fd104f80e0cd536a029483 

The asynchronous check works OK (throws exception if m_job_number is 0)

The synchronous checks throws an exception if the exit code of the shell command is non-zero.
Now the Shell.bad_shell test passes:
```
$ ctest -R Shell.bad_shell
Test project /home/nickw/work/sjef/sjef/build
    Start 53: Shell.bad_shell
1/1 Test #53: Shell.bad_shell ..................   Passed    0.06 sec

```
but lots of other tests now hang when this exception is thrown:
"test_sjef.spawn_many_dummy|test_sjef.restart|test_sjef.backend_cache|test_sjef.many_projects|test_sjef.dummy_backend|test_sjef.project_name_embedded_space|test_sjef.project_dir_embedded_space|test_sjef.sync_backend|test_sjef.xpath_search|test_sjef.corrupt_geometry_include|test_sjef_molpro.spawn_many_molpro|test_sjef_molpro.molpro_workflow|test_sjef_molpro.failure|test-sjef"

**Comment by pjknowles** [comment]
I can't get https://github.com/molpro/sjef/commit/3e48572f55131d2277fd104f80e0cd536a029483 - it seems not to be connected to a branch?

**Comment by nick-wilson** [comment]
Yes, it's on shell-capture-failures branch on fork in my repository
https://github.com/molpro/sjef/compare/master...nick-wilson:sjef:shell-capture-failures
